test/test_fcntl.rs

```diff
@@ -143,3 +144,16 @@ mod linux_android {
         assert_eq!(100, read(fd, &mut buf).unwrap());
     }
 }
+
+#[cfg(any(target_os = "linux", target_os = "macos", target_os = "android", target_os = "freebsd", target_os = "dragonfly", target_os = "openbsd"))]
+#[test]
+fn test_posix_fallocate() {
+    let tmp = NamedTempFile::new().unwrap();
+
+    let fd = tmp.as_raw_fd();
+    posix_fallocate(fd, 0, 100).unwrap();
```

Comment:
This test will fail if /tmp's filesystem doesn't support `posix_fallocate`.  ZFS, for example, does not.  POSIX stupidly requires that `posix_fallocate` return `EINVAL` in that case.  So you should skip the rest of the test if you get that errno.